allensdk/brain_observatory/behavior/behavior_session.py

```diff
@@ -83,6 +84,36 @@ def list_api_methods(self) -> List[Tuple[str, str]]:
         method_names = [m[0] for m in methods]
         return list(zip(method_names, docs))
 
+    def list_data_attributes_and_methods(self) -> List[str]:
+        """Convenience method for end-users to list attributes and methods
+        that can be called to access data for a BehaviorSession.
+
+        NOTE: Because BehaviorExperiment inherits from BehaviorSession
+        this method was also be available there.
+
+        Returns
+        -------
+        List[str]
+            A list of attributes and methods that end-users can access or call
+            to get data.
+        """
+        attrs_and_methods_to_ignore: set = {
+            "api",
+            "from_lims",
+            "from_nwb_path",
+            "LazyProperty",
+            "list_api_methods",
+            "list_data_attributes_and_methods"
+        }
+        attrs_and_methods_to_ignore.update(dir(ParamsMixin))
```

Comment:
I'm worried about this method getting out of date if, for some reason, we have to add new mixins to this class (or changes its inheritance more generally).

One option would be to use the `cls.__bases__` attribute to iterate over all of the basis classes of the current class, excluding their members (that assumes that we will never implement a mixin that provides a data-related method for the end user).

We could also add a unit test that verifies the output of this method. It would less be a unit test than a way for us to detect if we ever add a new method to this class that makes it into this list. We would then be forced to make the conscious choice whether the new method belongs in the list of data-related methods or not and update the unit test.

Reply:
I've added unit tests to check that if data changes, a test will fail.